Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPISidebarCocoa.mm

```diff
@@ -176,19 +206,24 @@
         return;
     }
 
+    if (!hasActiveUserGesture(*tab)) {
+        completionHandler(toWebExtensionError(nil, nil, @"open can only be called in response to a user action"));
```

Comment:
This is the wrong user gesture to check. The check needs to happen in the web process.

Do that via:
```
WebCore::UserGestureIndicator::processingUserGesture()
```

See `WebExtensionAPIPermissions::request()` for an example.

Reply:
Ahh I was wondering if this was the case, thank you!